test/rb/test_rb.py

```diff
@@ -726,6 +729,178 @@ def test_rb(self, nq, pattern_type, multiplier_type):
             self.assertEqual(circ_index, len(rb_purity_circs),
                              "Error: additional purity circuits exist")
 
+    @data(HGate(), SGate(), SdgGate(), XGate(), YGate(), ZGate())
+    def test_interleaved_randomized_benchmarking_seq_1q_clifford_gates(self, gate):
+        """interleaved 1Q Clifford gates in RB"""
+        rb_original_circs, _, rb_interleaved_circs = rb.randomized_benchmarking_seq(
+            1, [5], rb_pattern=[[0]], interleaved_elem=[gate])
+        # Verify the generated sequences
+        seed = 0
+        circ_index = 0
+        self.assertEqual(rb_original_circs[seed][circ_index].name,
```

Comment:
I feel like we can still do a bit more checking here besides just that the circuit name matches. The interleaved circuits just add the interleaved element between each section in the original rb circuit right, what about verifying that? We also can verify that there are the correct number of interleaved elements based on the length vector right.

For example, you can use `circuit.data()` to get a list of gates and qargs that we can use for matching and break things into sections across barriers. Then just check we're properly interleaving the element and we've got the expected number of interleaved elements.

Reply:
Thanks for the feedback.
 
Please note that there are already tests for interleaved RB as part of the `test_rb` function, in particular, we already do the comparison that you suggested in:
https://github.com/Qiskit/qiskit-ignis/blob/4c0369f5ba2db074968c847d14c86a9d90e49ccf/test/rb/test_rb.py#L703
In this PR, we only extended the possibility of the input interleaved elements from a `QuantumCircuit` to a `Gate` object.

Reply:
So why don't you call `self.compare_interleaved_circuit` here then? How do you know that in this change that interleaving a `Gate`  is working correctly? Sure, manual testing shows it works now, but the tests are also to ensure future changes don't regress working functionality. Like what if in a future PR someone makes a mistake and instead of interleaving the gate correctly when a `Gate` is the interleaved element it interleaves a duplicate of the original rb component or something weird like that. You want the tests to work. The tests here don't validate that the output circuit is interleaving the input `Gate` object correctly. For example, if you did something completely wrong like:
```diff
--- a/qiskit/ignis/verification/randomized_benchmarking/circuits.py
+++ b/qiskit/ignis/verification/randomized_benchmarking/circuits.py
@@ -152,6 +152,9 @@ def handle_interleaved_elem(interleaved_elem, rb_group):
             if isinstance(elem, (QuantumCircuit, Instruction)):
                 num_qubits = elem.num_qubits
                 qc = elem
+                if num_qubits == 1:
+                    from qiskit.circuit.library import SdgGate
+                    qc = SdgGate()
                 elem = rb_group.iden(num_qubits)
                 elem = elem.from_circuit(qc)
             if (isinstance(elem, qiskit.quantum_info.operators.symplectic.clifford.Clifford)
```
the tests would pass despite it being incorrectly interleaving a Sdg gate everywhere.

Reply:
Indeed - I added tests for `compare_interleaved_circuit` and `compare_cnotdihedral_circuit`